docs/src/modules/brandingTheme.ts

```diff
@@ -30,11 +30,11 @@ declare module '@material-ui/core/styles/createTypography' {
 }
 
 // TODO: enable this once types conflict is fixed
```

Comment:
Each subsequent module augmentation need to specify the original file the interface was declared:

> As I found via some helpful feedback on Gitter, in response to this S.O. issue, in order for declaration merging to work across multiple files, all subsequent declarations (after the first) need to specify the original file that the declared interface was created in.

-- https://github.com/microsoft/TypeScript/issues/33326

The problem is that `ButtonPropsVariantOverrides` isn't declared in `@material-ui/core/Button/index.d.ts` but `@material-ui/core/Button/Button.d.ts`. This also explains why re-exporting `ButtonPropsVariantOverrides` explicitly from `@material-ui/core/Button` worked because now `ButtonPropsVariantOverrides` is declared in the `index.d.ts` not in `Button.d.ts` i.e. `export *` does not change the location where an interface is declared in.

So in general, we want to export types that are subject to module augmentation explicitly from public paths. I'll do that in a separate PR.

Though it does show again that this theming based customization approach is fundamentally flawed. You have to make sure you have a single theme in your code base at which point you're better off creating a separate component. You get all the type-checking and discoverability for free and people don't need to learn a new API.

Reply:
For now, keep the module augmentation but augment `@material-ui/core/Button/Button` and leave a TODO.

Reply:
Good to know 👍.